File: dbflow/src/main/java/com/raizlabs/android/dbflow/sql/language/WhereBase.java
package com.raizlabs.android.dbflow.sql.language;

import androidx.annotation.NonNull;

import com.raizlabs.android.dbflow.sql.Query;

/**
 * Description: The base for a {@link Where} statement.
 */
public interface WhereBase<TModel> extends Query, Actionable {

    /**
     * @return The table of this query.
     */
    @NonNull
    Class<TModel> getTable();

    /**
     * @return The base Query object.
     */
    @NonNull
    Query getQueryBuilderBase();

}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/sql/language/WhereBase.kt
package com.raizlabs.android.dbflow.sql.language

import com.raizlabs.android.dbflow.sql.Query

/**
 * Description: The base for a [Where] statement.
 */
interface WhereBase<TModel> : Query, Actionable {

    /**
     * @return The table of this query.
     */
    val table: Class<TModel>

    /**
     * @return The base Query object.
     */
    val queryBuilderBase: Query

}
